contract_compassion/wizard/end_contract_wizard.py

```diff
@@ -64,22 +64,25 @@ def end_contract(self):
         self.ensure_one()
         contract = self.contract_id
         child = contract.child_id
-        hold = child.hold_id
 
-        if not self.has_new_hold:
-            # child will be inactive after this choice
-            self.hold_expiration_date = fields.Datetime.now()
-            child.write({'active': False})
-
-        # in all case current hold became inactive
-        hold.write({'active': False})
+        if self.has_new_hold:
+            self.env['icp.reservation'].create({
+                'name': 'single beneficiary reservation',
+                'expiration_date': self.end_date,

```

Comment:
In any case the reservation should not expire when the sponsorship is terminated. Otherwise we will never have it on hold.

I think end_date of contract is useless. You can modify it but the contract will be terminated immediately.